src/python/pants/base/deprecated.py

```diff
@@ -135,7 +135,8 @@ def warn_or_error(
   stacklevel: int = 3,
   frame_info: Optional[inspect.FrameInfo] = None,
   context: int = 1,
-  ensure_stderr: bool = False
+  ensure_stderr: bool = False,
+  print_warning: bool = True,
```

Comment:
This should be added to the docstring (the same for adding any new argument to a method with params in the docstring)! To me it's not immediately clear what this particular arg is supposed to represent.